intera_interface/src/intera_interface/limb.py

```diff
@@ -636,12 +639,8 @@ def ik_request(self, pose,
             return False
         limb_joints = {}
         # Check if result valid, and type of seed ultimately used to get solution
-        if (resp.result_type[0] > 0):
-            seed_str = {
-                        ikreq.SEED_USER: 'User Provided Seed',
-                        ikreq.SEED_CURRENT: 'Current Joint Angles',
-                        ikreq.SEED_NS_MAP: 'Nullspace Setpoints',
-                       }.get(resp.result_type[0], 'None')
+        if (resp.result_type[0] > 0
+                or (allow_collision and resp.result_type[0] == resp.IK_IN_COLLISION)):
```

Comment:
this logic is a little confusing.

If the result is valid, or if collision is allowed and the robot is in collision... er, this calls for a truth table!
```
A || (B && C)
Valid || (CollisionAllowed && InCollision)
False || (False && False) = Solution Invalid, collision not allowed, and not in collision = False
False || (False && True) = Solution Invalid, collision not allowed, and in collision = False
False || (True && False) = Solution Invalid, collision allowed, and not in collision = False
False || (True && True) = Solution Invalid, collision allowed, and in collision = True  <--- seems fishy
# True || short-circuits everything
True || (False && False) = Solution Valid, collision not allowed, and not in collision = True
True || (True && False) = Solution Valid, collision allowed, and not in collision = True
True || (False && True) = Solution Valid, collision not allowed, and in collision = True
True || (True && True) = Solution Valid, collision allowed, and in collision = True
```

Am I missing something here or is this wrong?